src/main/java/organice/logic/commands/DoneCommand.java

```diff
@@ -0,0 +1,92 @@
+package organice.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static organice.logic.commands.MatchCommand.match;
+
+import organice.model.Model;
+import organice.model.person.Donor;
+import organice.model.person.Nric;
+import organice.model.person.Patient;
+import organice.model.person.exceptions.PersonNotFoundException;
+
+public class DoneCommand extends Command {
+    public static final String COMMAND_WORD = "done";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finish the processing of patients and donors"
+            + "Parameters: ic/PATIENT NRIC ic/DONOR NRIC \n"
+            + "Example: " + COMMAND_WORD + " ic/s4512345A ic/s7711123C";
+
+    public static final String MESSAGE_SUCCESS = "Done processing the patient and donor";
+    public static final String MESSAGE_NOT_PROCESSED = "Donor or patient NRIC must be valid";
+
+    private String firstNricString;
+    private String secondNricString;
+
+    private Nric firstNric;
+    private Nric secondNric;
+
+    private Donor donor;
+    private Patient patient;
+
+    private Nric patientNric;
+    private Nric donorNric;
+
+
+    public DoneCommand(String firstNricString, String secondNricString) {
+        requireNonNull(firstNricString, secondNricString);
+        firstNric = new Nric(firstNricString);
+        secondNric = new Nric(secondNricString);
+    }
+
+    public boolean isValidDonorPatientPair(Nric firstNric, Nric secondNric, Model model) {
+        if (model.hasDonor(firstNric)) {
+            donorNric = firstNric;
+            donor = model.getDonor(donorNric);
+
+            patientNric = secondNric;
+            patient = model.getPatient(patientNric);
+        } else {
+            patientNric = firstNric;
+            patient = model.getPatient(patientNric);
+
+            donorNric = secondNric;
+            donor = model.getDonor(donorNric);
+        }
+        if (model.hasPatient(patientNric) && model.hasDonor(donorNric)
```

Comment:
I will recommend you to create a separate method to check if the task list is empty. So that it adheres to the single responsibility principle.